pkg/virtctl/imageupload/imageupload.go

```diff
@@ -203,7 +205,19 @@ func getHTTPClient(insecure bool) *http.Client {
 }
 
 func uploadData(uploadProxyURL, token string, file *os.File, insecure bool) error {
-	url := uploadProxyURL + uploadProxyURI
+
+	u, err := url.Parse(uploadProxyURL)
```

Comment:
can we put this logic that generates the url into a helper function? Then all we really need to do during unit testing is verify the output of the helper function rather than running a real http server. I think we just want to verify the right url string is generated.

Reply:
Done, please review

Reply:
Done, please review